### PR TITLE
[core/cling] modernize nullptr

### PR DESCRIPTION
**Pull request by ferdymercury**
by clang-tidy

### CODE REVIEW

**Review by pcanal — APPROVED** [review]
Thanks.

File: core/metacling/src/TClingClassInfo.cxx [hunk 1]
```diff
@@ -116,15 +116,15 @@ TClingClassInfo::TClingClassInfo(cling::Interpreter *interp, const char *name, b
...
+     fIsIter(false), fType(nullptr), fTitle(""), fOffsetCache(0)
 {
    Init(tag);
 }
 
 TClingClassInfo::TClingClassInfo(cling::Interpreter *interp,
                                  const Decl *D)
    : TClingDeclInfo(nullptr), fInterp(interp), fFirstTime(true), fDescend(false), fIterAll(kTRUE),
-     fIsIter(false), fType(0), fTitle(""), fOffsetCache(0)
+     fIsIter(false), fType(nullptr), fTitle(""), fOffsetCache(0)
```
Comment (pcanal) [inline core/metacling/src/TClingClassInfo.cxx:127]:
```suggestion
     fIsIter(false), fOffsetCache(0)
```
Given the warning from clang-tidy: https://travis-ci.org/github/root-project/root/jobs/772600580

**Review by pcanal — CHANGES_REQUESTED** [review]
See suggestions.

File: core/metacling/src/TClingClassInfo.cxx [hunk 2]
```diff
@@ -115,7 +115,7 @@ TClingClassInfo::TClingClassInfo(cling::Interpreter *interp, const char *name, b
 TClingClassInfo::TClingClassInfo(cling::Interpreter *interp,
                                  const Type &tag)
    : TClingDeclInfo(nullptr), fInterp(interp), fFirstTime(true), fDescend(false), fIterAll(kTRUE),
-     fIsIter(false), fType(nullptr), fTitle(""), fOffsetCache(0)
```
Comment (ferdymercury) [inline core/metacling/src/TClingClassInfo.cxx]: I thought clang only complained about fType, why removing also fTitle?

**Comment by pcanal** [comment]
>  redundant, already in base class

Note that the reason clang-tidy is recommending to remove the explicit setting of fType in the constructor is because it is already set in the class own's declaration (it is inlined in the declaration of the fType data member).

File: core/metacling/src/TClingClassInfo.cxx (hunk 2 above)
Comment (pcanal) [reply core/metacling/src/TClingClassInfo.cxx]: That is just me noticing that for an std::string the notation `fTitle("")` has the same effect as calling the default constructor.

**Comment by ferdymercury** [comment]
One comment: if we consider that these 'format' changes should not bloat the git history, we could add a `.git-blame-ignore-revs`, see more info:
https://www.moxio.com/blog/43/ignoring-bulk-change-commits-with-git-blame

**Review by pcanal — APPROVED** [review]
Thanks.

**Comment by ferdymercury** [comment]
@linev this one might fit with your latest merges (thanks!)

**Comment by linev** [comment]
Idea to wait with cling changes until llvm upgrade.

**Comment by ferdymercury** [comment]
> Idea to wait with cling changes until llvm upgrade.

Rebased now.

**Review by pcanal — APPROVED** [review]

**Pull request merged**